src/main/java/edu/umich/ctools/sectionsUtilityTool/Friend.java

```diff
@@ -0,0 +1,317 @@
+				M_log.debug("referrerUrl: " + referrerUrl);
+				M_log.debug("ksFileName: " + ksFileName);
+				M_log.debug("ksPwd: " + ksPwd);
+			}else {
+				M_log.error("Failed to load secure application properties from sectionsToolPropsSecure.properties for SectionsTool");
+			}
+			
+		}else {
+			M_log.error("File path for (sectionsToolPropsPathSecure.properties) is not provided");
+		}
+
+		Properties systemProps = System.getProperties();
+		String keyStoreType = (String) systemProps.get("javax.net.ssl.keyStoreType");
+		String trustStoreType = (String) systemProps.get("javax.net.ssl.trustStoreType");
+		if (keyStoreType != null && !KEYSTORETYPE_PKCS12.equals(keyStoreType)) // existing keyStoreType 
+		{
+			M_log.error(this + " init: existing settings of SSL keyStoreType mismatch: " + keyStoreType );
+			sslInitialized = false;
+		}
+		else if (trustStoreType != null  && !TRUSTSTORETYPE_JKS.equals(trustStoreType)) // existing trustStoreType
+		{
+			M_log.error(this + " init: existing settings of SSL trustStoretype mismatch: " + trustStoreType );
+			sslInitialized = false;
+		}
+		else
+		{	
+			// key store
+			systemProps.put("javax.net.ssl.keyStoreType", KEYSTORETYPE_PKCS12);
+			systemProps.put("javax.net.ssl.trustStoreType", TRUSTSTORETYPE_JKS);
+
+			if (ksFileName.isEmpty())
+			{
+				// log error for missing keystore file path
+				M_log.error(this + " init the ctools.friend.keystorefile path is not defined. ");
+			}
+			else
+			{
+				systemProps.put("javax.net.ssl.keyStore",ksFileName);
+			}
+
+			if (ksPwd.isEmpty())
+			{
+				// log error for missing keystore password
+				M_log.error(this + " init the ctools.friend.keystorepassword is not defined. ");
+			}
+			else
+			{
+				systemProps.put("javax.net.ssl.keyStorePassword",ksPwd);
+			}
+
+			// set system properties
+			System.setProperties(systemProps);
+
+			sslInitialized = true;
+		}
+	}	
+
+	/**
+	 * Returns to uninitialized state.
+	 */
+	public void destroy()
+	{
+		M_log.info(this +".destroy()");
+
+		if (sslInitialized)
+		{
+			// remove ssl system settings
+			// configure JSSE system properties: 
+			// http://fusesource.com/docs/broker/5.3/security/SSL-SysProps.html
+			Properties systemProps = System.getProperties();
+			//systemProps.remove("javax.net.debug");
+			//important: http://stackoverflow.com/questions/6680416/apache-cxf-exception-in-ssl-communication-sockettimeout
+			//java.lang.System.setProperty("sun.security.ssl.allowUnsafeRenegotiation", "true");
+			systemProps.remove("javax.net.ssl.keyStoreType");
+			systemProps.remove("javax.net.ssl.trustStoreType");
+			systemProps.remove("javax.net.ssl.keyStore");
+			systemProps.remove("javax.net.ssl.keyStorePassword");
+			System.setProperties(systemProps);
+
+			sslInitialized = false;
+		}
+	}	
+
+	/*
+	 * Check if an account exists
+	 * Return a single integer with the following (It all succeeds of fails)
+	 * @param email friend account Email
+	 * @return
+	 * -1 = bad e-mail address
```

Comment:
I would love to change it myself, however the result is part of a web service call that I have no control over. However, the numbers are given meaning when they are called from the servlet.